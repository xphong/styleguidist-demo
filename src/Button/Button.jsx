import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledButton = styled.button`
  display: flex;
  height: 38px;
  font-weight: 400;
  justify-content: center;
  align-items: center;
  padding: 0 25px;
  border-radius: 3px;
  background: #365DC9;
  color: #FFF;
  cursor: pointer;
`;

export default class Button extends React.PureComponent {
  static propTypes = {
    text: PropTypes.string
  };

  static defaultProps = {
    text: 'SAVE'
  };

  render() {
    return (
      <StyledButton>{this.props.text}</StyledButton>
    );
  }
}
