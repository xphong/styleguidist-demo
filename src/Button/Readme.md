```js
<Button dummyProp={123}/>
```