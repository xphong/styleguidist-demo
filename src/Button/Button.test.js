import React from 'react';
import { shallow } from 'enzyme';

import Button from './Button';

const createComponent = newProps => {
  const component = shallow(
    <Button {...newProps} />
  );

  return {
    component
  };
};

describe('Button component', () => {
  it('should render', () => {
    const { component } = createComponent();
    expect(component).toMatchSnapshot();
  });
});
