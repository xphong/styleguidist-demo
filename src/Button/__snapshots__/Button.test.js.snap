// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render 1`] = `
<p>
  replace me 
  123
</p>
`;
